test/filter_test.cc

```diff
@@ -75,7 +76,7 @@ int main(int argc, char* argv[]) {
 
   // Make sure we ran all of the tests
   const size_t count = test_reporter.GetCount();
-  const size_t expected = (argc == 2) ? std::stoul(argv[1]) : count;
+  const size_t expected = (argc == 2) ? std::atol(argv[1]) : count;
```

Comment:
i agree, however this is test code so we might be a little less strict if it allows them to run on a whole new platform.

i don't see any other workaround either for android. sadly their std implementation is broken in a few ways. perhaps we could define BENCHMARK_OS_ANDROID and have custom char -> unsigned long parsing just for them.

alternatively we could bring in stringstream here to parse the number instead, though that seems like overkill.